controllers/AdminController.php

```diff
@@ -294,23 +309,40 @@ public function actionInfo($id)
 
     /**
      * Switches to the given user for the rest of the Session.
+     * When no id is given, we switch back to the original admin user
+     * that started the impersonation.
      *
      * @param int $id
      *
      * @return string
      */
-    public function actionSwitch($id)
+    public function actionSwitch($id = null)
     {
-        $user = $this->findModel($id);
+        if(!Yii::$app->getModule('user')->enableImpersonateUser)
+            throw new ForbiddenHttpException(Yii::t('user', 'Impersonate user is disabled in the application configuration'));
+
+        $key = self::ORIGINAL_USER_SESSION_KEY;
```

Comment:
Yeah that's matter of style, but I don't like such read-only variables :)
It is even not that long :smile: I think it should be changed to usage of constant instead of variable.